ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -1942,10 +1942,10 @@ extension SignalProducerType {
 		// out of scope. This lets us know when we're supposed to dispose the
 		// underlying producer. This is necessary because `struct`s don't have
 		// `deinit`.
-		let token = DeallocationToken()
+		let token = NSObject()
```

Comment:
I think we should use a `Lifetime` here directly. `NSObject.lifetime` is available only on Darwin platforms.